components/aws/sagemaker/common/_utils.py

```diff
@@ -113,17 +122,49 @@ def print_logs_for_job(cw_client, log_grp, job_name):
         logging.error(e)
 
 
-def get_sagemaker_client(region, endpoint_url=None):
+def get_boto3_session(region, role_arn=None):
+    """Creates a boto3 session, optionally assuming a role"""
+
+    # By default return a basic session
+    if role_arn is None:
+        return Session()
```

Comment:
if role_arn is specified region is set in the session object, so lets set it when role_arn is None as well and avoid setting it while creating the client

Reply:
pending work on this comment